helix-term/src/commands.rs

```diff
@@ -2380,7 +2412,20 @@ fn jumplist_picker(cx: &mut Context) {
     impl ui::menu::Item for JumpMeta {
         type Data = ();
 
-        fn format(&self, _data: &Self::Data) -> Row {
+        fn format<'a>(&self, _data: &Self::Data, icons: Option<&'a Icons>) -> Row {
+            // Get the filetype icon, or a "file" icon for scratch buffers
+            let icon = icons.and_then(|icons| {
+                self.path
+                    .as_ref()
+                    .and_then(|path| icons.icon_from_path(path))
+                    .or_else(|| {
+                        icons
+                            .symbol_kind
+                            .as_ref()
+                            .and_then(|symbol_kind_icons| symbol_kind_icons.get("file"))
+                    })
+            });
```

Comment:
This could maybe be a helper on `Icons` since it seems to be used in both `JumpMeta` and `BufferMeta`?

Reply:
Good idea, I modified the `Icons::icon_from_path()` function to factorize this.